test/unit/tasks/publishing_api_test.rb

```diff
@@ -0,0 +1,384 @@
+require "test_helper"
+
+class PublishingApiRake < ActiveSupport::TestCase
+  extend Minitest::Spec::DSL
+
+  teardown do
+    task.reenable if defined?(task) # without this, calling `invoke` does nothing after first test
```

Comment:
It's a shame this doesn't get applied to the `republish` set of tests, though it could if we defined a `let(:task)` in there. Can we do that?

Reply:
Agreed so I've tried to refactor using `let`, with not much success. I'm open to suggestions though 😄

Reply:
I've had a go here: https://github.com/alphagov/whitehall/commit/fe6d249da76630a37862f1d8a33bf442e7352d5d
It does lead to _slightly_ more lines of code, but I think this is worth it for the increased readability.

The rest of the PR looks great @1pretz1, but think it's worth making this one last tweak before merge ✅

Reply:
Sure thing, I've split them into their own separate tests. I was in two minds when writing them as it did seem like a compromise on consistency when trying to DRY up the code.